packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1453,14 +1455,152 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
+    return RenderStack.getIntrinsicDimension(firstChild, (RenderBox child) => child.getMinIntrinsicWidth(height));
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return RenderStack.getIntrinsicDimension(firstChild, (RenderBox child) => child.getMaxIntrinsicWidth(height));
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    return RenderStack.getIntrinsicDimension(firstChild, (RenderBox child) => child.getMinIntrinsicHeight(width));
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return RenderStack.getIntrinsicDimension(firstChild, (RenderBox child) => child.getMaxIntrinsicHeight(width));
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    assert(!debugNeedsLayout);
+    double result;
+    RenderBox child = firstChild;
+    while (child != null) {
+      assert(!child.debugNeedsLayout);
+      final StackParentData childParentData = child.parentData as StackParentData;
+      double candidate = child.getDistanceToActualBaseline(baseline);
+      if (candidate != null) {
+        candidate += childParentData.offset.dy;
+        if (result != null) {
+          result = math.min(result, candidate);
+        } else {
+          result = candidate;
+        }
+      }
+      child = childParentData.nextSibling;
+    }
+    return result;
+  }
+
+  @override
+  bool get sizedByParent => true;
+
+  @override
+  void performResize() {
+    size = constraints.biggest;
+    assert(size.isFinite);
+  }
+
+  @override
+  void performLayout() {
+    final BoxConstraints nonPositionedConstraints = BoxConstraints.tight(constraints.biggest);
+
+    RenderBox child = firstChild;
+    while (child != null) {
+      final StackParentData childParentData = child.parentData as StackParentData;
+      child.layout(nonPositionedConstraints, parentUsesSize: true);
+      child = childParentData.nextSibling;
+    }
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, { Offset position }) {
+    return defaultHitTestChildren(result, position: position);
+  }
+
+  @protected
+  void paintStack(PaintingContext context, Offset offset) {
+    RenderBox child = firstChild;
+    while (child != null) {
+      final StackParentData childParentData = child.parentData as StackParentData;
+      context.paintChild(child, childParentData.offset + offset);
+      child = childParentData.nextSibling;
+    }
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    RenderBox child = firstChild;
+    while (child != null) {
+      final StackParentData childParentData = child.parentData as StackParentData;
+      context.paintChild(child, childParentData.offset + offset);
+      child = childParentData.nextSibling;
+    }
+  }
+
+  @override
+  void visitChildrenForSemantics(RenderObjectVisitor visitor) {
+    // Semantics should focus on page content first instead of modal barrier.
```

Comment:
Will this mess up hit testing on e.g. Android? I would expect that on Android whenever you try to touch anything in the alert to a11y focus it, you’ll actually focus the moral barrier now...